test/integration/newapp_test.go

```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```

Comment:
P.S. Another try to explain my position: I'm fixing the fact that tests create files and don't remove them. For this task it's irrelevant where these files are created, it's a different story and I don't want to deal with that. If someone wants to fix it, let him do it in a separate PR.